.github/workflows/ci.yml

```diff
@@ -0,0 +1,23 @@
+# This can be run manually like so:
+#
+#     gh workflow run ci.yml --ref <git_branch_name>
+#
+name: Continuous Integration
+on:
+  pull_request:
+  push:
+  workflow_dispatch:
+jobs:
+  build:
+    name: nix-build
+    runs-on: self-hosted
```

Comment:
@nixinator Can we setup self-hosted CI for this repo? Does that automatically give us nix cache? Otherwise `runs-on: ubuntu-latest` may be sufficient.

Reply:
were using herc-ci ,i'll give you setup details soon.

Reply:
i need to find out how to do the tests you need in herc-ci. So bear with me. I really want to go the herc route because the runners are some much more nix compatible, self hosted github runners are PITA.

Reply:
Sure, I think hercules-ci is generally better. Can the nix cache be shared with MLabs folks as well?